### PR TITLE
.golangci.yml: Use depguard to blacklist github.com/pkg/errors

### LINKED ISSUE #19804: Pick one error package
We currently use three different error packages:

* https://golang.org/pkg/errors/
* https://github.com/cockroachdb/errors
* github.com/pkg/errors

We should pick 1 and stick with it :)

The cockroach db one seems to provide the features of all the others so maybe that's the one to go for.

### PR DESCRIPTION
**Pull request by indradhanush**
Link to build where the `depguard` check works: https://buildkite.com/sourcegraph/sourcegraph/builds/99028#4c7d4d4b-9eb9-4216-aefc-295b233d3456/466-467

I've since reverted the commits that added `github.com/pkg/errors` in this branch.

Fixes #19804.
Follow up on #21684 

COREAPP-65 #done

### CODE REVIEW

**Review by efritz — APPROVED** [review]

**Comment by camdencheek** [comment]
Nice! glad this worked so well for ya

File: .golangci.yml [hunk 1]
```diff
@@ -15,6 +16,10 @@ linters:
     - unused
 
 linters-settings:
+  depguard:
+    list-type: blacklist
```
Comment (unknwon) [inline .golangci.yml:20]: nit: we've started trying to use "blocklist" and "allowlist" in our codebase.

Comment (efritz) [reply .golangci.yml:20]: I don't think we have the authority to change this in golangci-lint, though...

Comment (unknwon) [reply .golangci.yml:20]: Oh I thought the "list-type" is user-defined 😂

**Pull request merged**